src/host/kvpair.rs

```diff
@@ -88,24 +71,31 @@ impl<const DEPTH: usize> MongoMerkle<DEPTH> {
     ) -> Result<Option<MerkleRecord>, mongodb::error::Error> {
         let dbname = Self::get_db_name();
         let cname = self.get_collection_name();
-        let collection = get_collection::<MerkleRecord>(&self.client, dbname, cname)?;
-        let mut filter = doc! {};
-        filter.insert("index", index);
-        filter.insert("hash", bytes_to_bson(hash));
-        collection.find_one(filter, None)
+        let cache_key = cname.clone() + &index.to_string() + &hex::encode(hash);
+        let mut cache = MERKLE_CACHE.lock().unwrap();
+        if let Some(record) = cache.get(&cache_key) {
+            Ok(Some(record.clone()))
+        } else {
+            let collection = db::get_collection::<MerkleRecord>(dbname, cname)?;
+            let mut filter = doc! {};
+            filter.insert("index", index);
+            filter.insert("hash", bytes_to_bson(hash));
+            collection.find_one(filter, None)
```

Comment:
Cache miss, so we load it from db, then do we need insert it into cache?

Reply:
Yes, you are right. I have fixed it.